### PR TITLE
[config/console] Support add/del console port setting commands

### PR DESCRIPTION
**Pull request by Blueve**
**- What I did**

Support `console add` and `console del` commands under config command group. (More detail in HLD: https://github.com/Azure/SONiC/blob/master/doc/console/SONiC-Console-Switch-High-Level-Design.md#33121-adddel-console-port)
Added UT for commands.

**- How I did it**

1. Add new `console.py` file to support `config console` commands
1. Add UTs for console.py

**- How to verify it**

1. UTs are all passed
1. Tested on a physical SONiC DUT, detail output below:
Initial state
```bash
admin@sonic:~$ sudo show line
  Line    Actual/Configured Baud    PID    Start Time    Device
------  ------------------------  -----  ------------  --------
     0      9600/-                                            -
     1      9600/9600                                   switch1
     2      9600/-                                            -
     3      9600/-                                            -
```

Error Handling
```bash
admin@sonic:~$ sudo config console add 1
Usage: config console add [OPTIONS] <line_number>
Try "config console add -h" for help.

Error: Missing option "--baud" / "-b".

admin@sonic:~$ sudo config console add 1 --baud 9600
Usage: config console add [OPTIONS] <line_number>
Try "config console add -h" for help.

Error: Trying to add console port setting, which is already exists

admin@sonic:~$ sudo config console add 2 --baud 9600 --flowcontrol --devicename switch0
Usage: config console add [OPTIONS] <line_number>
Try "config console add -h" for help.

Error: Given device name switch0 has been used. Please enter a valid device name or remove the existing one !!

admin@sonic:~$ sudo config console del 3
Usage: config console del [OPTIONS] <line_number>
Try "config console del -h" for help.

Error: Trying to delete console port setting, which is not present
```

Add new console port setting
```bash
admin@sonic:~$ sudo config console add 0 --baud 9600 --flowcontrol --devicename switch0
admin@sonic:~$ sudo show line
  Line    Actual/Configured Baud    PID    Start Time    Device
------  ------------------------  -----  ------------  --------
     0      9600/9600                                   switch0
     1      9600/9600                                   switch1
     2      9600/-                                            -
     3      9600/-                                            -
```

Remove existing console port setting
```bash
admin@sonic:~$ sudo config console del 0
admin@sonic:~$ sudo show line
  Line    Actual/Configured Baud    PID    Start Time    Device
------  ------------------------  -----  ------------  --------
     0      9600/-                                            -
     1      9600/9600                                   switch1
     2      9600/-                                            -
     3      9600/-                                            -
```

**- Previous command output (if the output of a command-line utility has changed)**
N/A

**- New command output (if the output of a command-line utility has changed)**
N/A

### CODE REVIEW

File: config/console.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    ctx = click.get_current_context()
+    data = config_db.get_entry(table, linenum)
+    if data:
+        ctx.fail("Trying to add console port setting, which is already exists.")
+    else:
+        console_entry = { dataKey1: baud }
+        console_entry[dataKey2] = "1" if flowcontrol else "0"
+
+        if devicename:
+            if isExistingSameDevice(config_db, devicename, table):
```
Comment (yxieca) [inline config/console.py:42]: Could it be that user are updating an existing configuration and give device name again?

File: tests/console_test.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    @classmethod
+    def setup_class(cls):
+        print("SETUP")
+    
+    def test_console_add_exists(self):
+        runner = CliRunner()
+        db = Db()
+        db.cfgdb.set_entry("CONSOLE_PORT", 1, { "baud_rate" : "9600" })
+
+        # add a console setting which the port has exists
```
Comment (yxieca) [inline tests/console_test.py:21]: s/has//

File: tests/console_test.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        result = runner.invoke(config.config.commands["console"].commands["add"], ["1", '--baud', "9600"], obj=db)
+        print(result.exit_code)
+        print(sys.stderr, result.output)
+        assert result.exit_code != 0
+        assert "Trying to add console port setting, which is already exists." in result.output
```
Comment (yxieca) [inline tests/console_test.py:26]: Why is this a failure? Do we want to support update configuration of an existing line?

File: config/console.py (hunk 1 above)
Comment (Blueve) [reply config/console.py:42]: We could, but not thru. this command, see HLD: https://github.com/Azure/SONiC/blob/master/doc/console/SONiC-Console-Switch-High-Level-Design.md#33122-update-remote-device-name-for-a-console-port

File: tests/console_test.py (hunk 3 above)
Comment (Blueve) [reply tests/console_test.py:26]: Yes, we want to support update configuration for an existing line.

**Review by yxieca — DISMISSED** [review]

**Review by yxieca — APPROVED** [review]

**Pull request merged**